tests/kafkatest/services/util.py

```diff
@@ -0,0 +1,63 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from contextlib import contextmanager
+from ducktape.utils.util import wait_until
+import subprocess
+
+def ssh_output(account, cmd, **kwargs):
+    return "\n".join(list(account.ssh_capture(cmd, **kwargs)))
+
+@contextmanager
+def monitor_log(node, log):
+    """
+    Context manager that returns an object that helps you wait for events to
+    occur in a log. This checks the size of the log at the beginning of the
+    block and makes a helper object available with convenience methods for
+    checking or waiting for a pattern to appear in the log. This will commonly
+    be used to start a process, then wait for a log message indicating the
+    process is in a ready state.
+
+    See LogMonitor for more usage information.
+    """
+    try:
+        offset = int(ssh_output(node.account, "wc -c %s" % log).split()[0])
+    except subprocess.CalledProcessError:
+        offset = 0
+    yield LogMonitor(node, log, offset)
+
+class LogMonitor(object):
```

Comment:
This is cool and surprisingly simple.
Is there any reason why you're doing a byte-based vs line-based approach as far as "offsets" are concerned?